src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -161,7 +161,7 @@ public static function getSubscribedEvents(): array
 
     private function mapQueryString(Request $request, string $type, MapQueryString $attribute): ?object
     {
-        if (!$data = $request->query->all()) {
+        if (!($data = $request->query->all()) && $attribute->metadata->hasDefaultValue()) {
```

Comment:
Are you missing to do the same for `mapRequestPayload`?
```php
if ('' === ($data = $request->getContent()) && $attribute->metadata->hasDefaultValue()) {
    return null;
}
```
Then it'll fail with a better error as well

Reply:
@yceruto I got quite busy for last week. I've noticed that `RequestPayloadValueResolver` should also be checking whether parameter is nullable or not. That way there appears to be no behavior change. Tests includes cases when parameter is nullable or has default value.